src/Runner.Worker/ActionManager.cs

```diff
@@ -503,39 +503,44 @@ private async Task DownloadRepositoryActionAsync(IExecutionContext executionCont
                 string apiUrl = GetApiUrl(executionContext);
                 string archiveLink = BuildLinkToActionArchive(apiUrl, repositoryReference.Name, repositoryReference.Ref);
                 Trace.Info($"Download archive '{archiveLink}' to '{destDirectory}'.");
-                await DownloadRepositoryActionAsync(executionContext, archiveLink, destDirectory);
+                var downloadDetails = new ActionDownloadDetails(archiveLink, ConfigureAuthorizationFromContext);
+                await DownloadRepositoryActionAsync(executionContext, downloadDetails, destDirectory);
                 return;
             }
             else
             {
                 string apiUrl = GetApiUrl(executionContext);
 
                 // URLs to try:
-                var archiveLinks = new List<string> {
+                var downloadAttempts = new List<ActionDownloadDetails> {
                     // A built-in action or an action the user has created, on their GHES instance
                     // Example:  https://my-ghes/api/v3/repos/my-org/my-action/tarball/v1
-                    BuildLinkToActionArchive(apiUrl, repositoryReference.Name, repositoryReference.Ref),
-
-                    // A community action, synced to their GHES instance
-                    // Example:  https://my-ghes/api/v3/repos/actions-community/some-org-some-action/tarball/v1
-                    BuildLinkToActionArchive(apiUrl, $"actions-community/{repositoryReference.Name.Replace("/", "-")}", repositoryReference.Ref)
+                    new ActionDownloadDetails(
+                        BuildLinkToActionArchive(apiUrl, repositoryReference.Name, repositoryReference.Ref),
+                        ConfigureAuthorizationFromContext),
+
+                    // The same action, on GitHub.com
+                    // Example:  https://api.github.com/repos/my-org/my-action/tarball/v1
+                    new ActionDownloadDetails(
+                        BuildLinkToActionArchive(_dotcomApiUrl, repositoryReference.Name, repositoryReference.Ref),
+                        configureAuthorization: (e,h) => { /* no authorization for dotcom */ })
                 };
 
-                foreach (var archiveLink in archiveLinks)
+                foreach (var downloadAttempt in downloadAttempts)
```

Comment:
makes sense after. after david's changes are in we can delete the for loop and converge back down to single code path for dotcom and ghes